pylint/checkers/non_ascii_names.py

```diff
@@ -0,0 +1,264 @@
+            and self._non_ascii_rgx_compiled.match(check_string) is None
+        ):
+            # Note that we require the `.isascii` method as it is fast and
+            # handles the complexities of unicode, so we can use simple regex.
+            self._raise_name_warning(node, node_type, name)
+            current_state = False
+
+        node._is_ascii_only = current_state  # pylint: disable=protected-access
+
+    @pylint.checkers.utils.check_messages("non-ascii-name")
+    def visit_module(self, node: nodes.Module) -> None:
+        self._check_name("file", node.name.split(".")[-1], node)
+
+    @pylint.checkers.utils.check_messages("non-ascii-name")
+    def visit_functiondef(
+        self, node: Union[nodes.FunctionDef, nodes.AsyncFunctionDef]
+    ) -> None:
+        self._check_name("function", node.name, node)
+
+        # Check argument names
+        arguments = node.args
+
+        # Check position only arguments
+        if arguments.posonlyargs is not None:
+            for pos_only_arg in arguments.posonlyargs:
+                self._check_name("argument", pos_only_arg.name, pos_only_arg)
+
+        # Check "normal" arguments
+        args = arguments.args
+        if args is not None:
+            self._recursive_check_names(args)
+
+        # Check key word only arguments
+        if arguments.kwonlyargs is not None:
+            for kwarg in arguments.kwonlyargs:
+                self._check_name("argument", kwarg.name, kwarg)
+
+    visit_asyncfunctiondef = visit_functiondef
+
+    @pylint.checkers.utils.check_messages("non-ascii-name")
+    def visit_global(self, node: nodes.Global) -> None:
+        for name in node.names:
+            self._check_name("const", name, node)
+
+    @pylint.checkers.utils.check_messages("non-ascii-name")
+    def visit_assignname(self, node: nodes.AssignName) -> None:
+        """check module level assigned names"""
+        # The NameChecker from which this Checker originates knows a lot of different
+        # versions of variables, i.e. constants, inline variables etc.
+        # To simplify we use only `variable` here, as we don't need to apply different
+        # rules to different types of variables.
+        frame = node.frame()
+        assign_type = node.assign_type()
+        if isinstance(assign_type, nodes.Comprehension):
+            # called inlinevar in NamesChecker
+            self._check_name("variable", node.name, node)
+        elif isinstance(frame, nodes.Module):
+            self._check_name("variable", node.name, node)
+        elif isinstance(frame, nodes.FunctionDef):
+            if not hasattr(node, "_is_ascii_only"):
+                # only check if not already done
+                self._check_name("variable", node.name, node)
+        elif isinstance(frame, nodes.ClassDef):
+            self._check_name("attr", node.name, node)
+        else:
+            # Just to make sure we check EVERYTHING (!)
+            self._check_name("variable", node.name, node)
```

Comment:
We seem to covering this according to `coveralls`. Might be worth adding a `breakpoint()` locally and seeing what type of `node` triggers this so we can add another case for it.

Reply:
I will simplify that.